bundles/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/SatelBinding.java

```diff
@@ -0,0 +1,240 @@
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "Satel Refresh Service";
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void execute() {
+		if (!this.satelModule.isInitialized()) {
+			logger.debug("Module not initialized yet, skipping refresh");
+			return;
+		}
+
+		// for first refresh after connecting to the module, make full refresh
+		// after that check what has changed since last refresh
+		logger.trace("Sending 'get new states' command");
+		this.satelModule
+				.sendCommand(NewStatesCommand.buildMessage(this.satelModule.getIntegraType() == IntegraType.I256_PLUS));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+		logger.trace("Binding configuration updated");
+
+		if (config == null) {
+			return;
+		}
+
+		this.refreshInterval = getLongValue(config, "refresh", 10000);
+		this.userCode = getStringValue(config, "user_code", null);
+
+		int timeout = getIntValue(config, "timeout", 5000);
+		String host = getStringValue(config, "host", null);;
+		if (StringUtils.isNotBlank(host)) {
+			this.satelModule = new Ethm1Module(host, getIntValue(config, "port", 7094), timeout,
+					(String) config.get("encryption_key"));
+		} else {
+			this.satelModule = new IntRSModule((String) config.get("port"), timeout);
+		}
+
+		this.satelModule.addEventListener(this);
+		this.satelModule.open();
+		setProperlyConfigured(true);
+		logger.trace("Binding properly configured");
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		if (!isProperlyConfigured()) {
+			logger.warn("Binding not properly configured, exiting");
+			return;
+		}
+
+		if (!this.satelModule.isInitialized()) {
+			logger.debug("Module not initialized yet, ignoring command");
+			return;
+		}
+
+		for (SatelBindingProvider provider : providers) {
+			SatelBindingConfig itemConfig = provider.getItemConfig(itemName);
+			if (itemConfig != null) {
+				logger.trace("Sending internal command for item {}: {}", itemName, command);
+				SatelMessage message = itemConfig.handleCommand(command, this.satelModule.getIntegraType(),
+						this.userCode);
+				if (message != null) {
+					this.satelModule.sendCommand(message);
+				}
+				break;
+			}
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void incomingEvent(SatelEvent event) {
+		logger.trace("Handling incoming event: {}", event);
+
+		// refresh all states that have changed
+		if (event instanceof NewStatesEvent) {
+			NewStatesEvent nse = (NewStatesEvent) event;
+			List<SatelMessage> commands = getRefreshCommands();
+			for (SatelMessage message : commands) {
+				if (nse.isNew(message.getCommand())) {
+					this.satelModule.sendCommand(message);
+				}
+			}
+		}
```

Comment:
NewStatesEvent is generated in response of NewStatesCommand and returns list of "things" that has changed since last check. For these "things" and only them (with some exceptions) we must prepare commands to read current state. Basically this is the flow implemented here.